### PR TITLE
MVP CLI Proposal implementation: proof/verify functionality 

### LINKED ISSUE #158: MVP CLI Proposal implementation: proof/verify functionality
Continuation of implementation of issue #70

- [ ] CLI command to return a proof at an index
- [ ] CLI command to prove the RLN proof
- [ ] CLI command to verify the RLN proof
- [ ] CLI command to generate the RLN proof
- [ ] CLI command to verify with roots

### PR DESCRIPTION
**Pull request by tyshko-rostyslav**
Closes #158

### CODE REVIEW

**Comment by rymnc** [comment]
@tyshko-rostyslav can we rebase this pr to master?

**Comment by tyshko-rostyslav** [comment]
> @tyshko-rostyslav can we rebase this pr to master?

I guess merge is better

File: rln-cli/src/main.rs [hunk 1]
```diff
@@ -93,7 +93,7 @@ fn main() -> Result<()> {
         Some(Commands::DeleteLeaf { index }) => {
             state
                 .rln
-                .ok_or(Report::msg("no RLN instance initialized"))?
+                .ok_or(Report::msg("no RLN initialized"))?
```
Comment (rymnc) [inline rln-cli/src/main.rs:96]:
```suggestion
                .ok_or(Report::msg("no RLN instance initialized"))?
```

File: rln-cli/src/main.rs [hunk 2]
```diff
@@ -105,6 +105,49 @@ fn main() -> Result<()> {
...
+            let input_data = File::open(&file)?;
+            state
+                .rln
+                .ok_or(Report::msg("no RLN instance initialized"))?
+                .verify(input_data)?;
+            Ok(())
+        }
+        Some(Commands::GenerateProof { input, output }) => {
+            let input_data = File::open(&input)?;
+            let output_data = File::open(&output)?;
```
[the remark this answers is not in the document]
Comment (tyshko-rostyslav) [reply rln-cli/src/main.rs]: Done

File: rln-cli/src/main.rs (hunk 1 above)
Comment (tyshko-rostyslav) [reply rln-cli/src/main.rs]: Fixed

File: rln-cli/src/main.rs [hunk 3]
```diff
@@ -105,6 +105,49 @@ fn main() -> Result<()> {
...
+            let output_data = File::open(&file)?;
+            state
+                .rln
+                .ok_or(Report::msg("no RLN instance initialized"))?
+                .get_proof(*index, output_data)?;
+            Ok(())
+        }
+        Some(Commands::Prove { input, output }) => {
+            let input_data = File::open(&input)?;
+            let output_data = File::open(&output)?;
```
[the remark this answers is not in the document]
Comment (tyshko-rostyslav) [reply rln-cli/src/main.rs]: Done

File: rln-cli/src/main.rs [hunk 4]
```diff
@@ -105,6 +105,49 @@ fn main() -> Result<()> {
                 .get_root(writer)?;
             Ok(())
         }
+        Some(Commands::GetProof { index, file }) => {
+            let output_data = File::open(&file)?;
```
[the remark this answers is not in the document]
Comment (tyshko-rostyslav) [reply rln-cli/src/main.rs]: Done

**Review by rymnc — APPROVED** [review]
LGTM

**Pull request merged**